src/getCartReceptors.C

```diff
@@ -127,21 +135,81 @@ void MeshBlock::getCartReceptors(CartGrid *cg,parallelComm *pc)
 		      pmap[cg->proc_id[c]]=1;
 		      dataPtr->next=(INTEGERLIST2 *) malloc(sizeof(INTEGERLIST2));
 		      dataPtr=dataPtr->next;
-		      dataPtr->realData=(double *) malloc(sizeof(double)*3);
-		      dataPtr->intData=(int *) malloc(sizeof(int)*3);
+          dataPtr->intDataSize=4;
+          dataPtr->realDataSize=4;
+          dataPtr->realData=(double *) malloc(sizeof(double)*dataPtr->realDataSize);
+          dataPtr->intData=(int *) malloc(sizeof(int)*dataPtr->intDataSize);
 		      dataPtr->intData[0]=cg->proc_id[c];
 		      dataPtr->intData[1]=cg->local_id[c];
 	        dataPtr->intData[2]=itm;
-		      dataPtr->intDataSize=3;
-		      dataPtr->realDataSize=1;
+	        dataPtr->intData[3]=0;
 		      nsearch+=1;
 
-			  for(kk=0;kk<3;kk++)
-			    dataPtr->realData[kk]=xtm[kk];
+          for(kk=0;kk<dataPtr->realDataSize-1;kk++)
+            dataPtr->realData[kk]=xtm[kk];
+
+          dataPtr->realData[dataPtr->realDataSize-1]=vol;
 
 		      dataPtr->next=NULL;
 		    }
 		}
+
+    // loop over all nodes
```

Comment:
Other than indexing offsets, the logic here is similar to what's in the cell loop right? Why not create a function that implements this once and pass appropriate starting indices?

Reply:
@sayerhs I have addresed this in 72c3a30